### PR TITLE
Custom formatters design

### PR DESCRIPTION
**Pull request by domenkozar**
@p1c2u I've sketched out how this would look.

A few comments:

- I'm using http://attrs.org/en/stable/index.html which is IMHO still better than dataclasses and works across all python versions
- I've merged validation/casting into one object to make sure both are supplied
- [x] tests

### CODE REVIEW

**Comment by p1c2u** [comment]
Merging validation/casting to one object with interface is good idea. I will use `attires` with models too.

File: openapi_core/schema/schemas/models.py [hunk 1]
```diff
@@ -22,6 +24,11 @@
 
 log = logging.getLogger(__name__)
 
+@attr.s
+class StringFormat(object):
```
Comment (p1c2u) [inline openapi_core/schema/schemas/models.py]: I would call it just `Format`

File: openapi_core/schema/schemas/models.py [hunk 2]
```diff
@@ -22,6 +24,11 @@
...
+    format = attr.ib()
```
Comment (p1c2u) [inline openapi_core/schema/schemas/models.py]: `unmarshal`?

File: openapi_core/schema/schemas/models.py [hunk 3]
```diff
@@ -78,6 +78,7 @@ def __init__(
 
         self._all_required_properties_cache = None
         self._all_optional_properties_cache = None
+        self.custom_formatters = None
```
Comment (p1c2u) [inline openapi_core/schema/schemas/models.py]: We shouldn't store custom formatters inside Schema object.

**Comment by domenkozar** [comment]
@p1c2u addressed your review comments :)

**Comment by domenkozar** [comment]
@p1c2u tests added.

**Comment by domenkozar** [comment]
@p1c2u is this good to go?

**Pull request closed without merging**